### PR TITLE
Add `path` on GCS targets in `google_storage_transfer_job`

### LINKED ISSUE #8704: google_storage_transfer_job add support for path in gcp data sink
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

We would like support for the path parameter in the gcp data sink as described in https://cloud.google.com/storage-transfer/docs/reference/rest/v1/TransferSpec#GcsData

We have a lot of source buckets which we want to aggregate in to one of destination buckets depending of the backup policy that is needed .

### New or Affected Resource(s)

* google_storage_transfer_job

### Potential Terraform Configuration

```tf
resource "google_storage_transfer_job" "s3-bucket-nightly-backup" {
  description = "Nightly backup of S3 bucket"
  project     = var.project
  transfer_spec {
    gcs_data_sink {
      bucket_name = google_storage_bucket.s3-backup-bucket.name
      path ="destinationpath/"
    }
  }
}
```

### References

adding support for this parameter is a lot like adding support for the intevalPeriod https://github.com/hashicorp/terraform-provider-google/issues/8578

Both can be handeld together

* #0000

### PR DESCRIPTION
**Pull request by bambuchaAdm**
Fix #8704 

It is my first contribution and i am little concern about style in documentation. It was copied out form REST api docs, but could be little confusing on usage. What do you think ?

### CODE REVIEW

File: google/resource_storage_transfer_job.go [hunk 1]
```diff
@@ -320,6 +320,13 @@ func gcsDataSchema() *schema.Resource {
 				Type:        schema.TypeString,
 				Description: `Google Cloud Storage bucket name.`,
 			},
+			"path": {
+				Optional:     true,
+				Type:         schema.TypeString,
+				Description:  `Google Cloud Storage path in bucket to transfer`,
+				Default:      "",
```
Comment (ScottSuarez) [inline google/resource_storage_transfer_job.go:327]: we should remove the default value here. Ideally we won't send this value if not set by the user. 

If the value is returned by the api after creation use `Computed: true`

File: google/resource_storage_transfer_job.go [hunk 2]
```diff
@@ -697,12 +704,14 @@ func expandGcsData(gcsDatas []interface{}) *storagetransfer.GcsData {
 	gcsData := gcsDatas[0].(map[string]interface{})
 	return &storagetransfer.GcsData{
 		BucketName: gcsData["bucket_name"].(string),
+		Path:       gcsData["path"].(string),
```
Comment (ScottSuarez) [inline google/resource_storage_transfer_job.go:707]: conditionally set this based on whether it exists in the map

File: google/resource_storage_transfer_job.go [hunk 3]
```diff
@@ -697,12 +704,14 @@ func expandGcsData(gcsDatas []interface{}) *storagetransfer.GcsData {
...
 	}
 }
 
 func flattenGcsData(gcsData *storagetransfer.GcsData) []map[string]interface{} {
 	data := map[string]interface{}{
 		"bucket_name": gcsData.BucketName,
+		"path":        gcsData.Path,
```
Comment (ScottSuarez) [inline google/resource_storage_transfer_job.go:714]: This is fine if we are using `Computed: true` in the schema definition.

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

**Comment by bambuchaAdm** [comment]
Property marked as computed. Is this mark would prevent engine from detect difference if this variable would be not specified in resource ? I also made little amendment in first commit message to reference original issue.

**Comment by bambuchaAdm** [comment]
@ScottSuarez Would you like to make again review on code after requested fixes ?

**Comment by ScottSuarez** [comment]
ah this has escaped me, let me take a looksee. sorry about that /gcbrun

**Comment by ScottSuarez** [comment]
gosh sorry about the delay ! let me upstream the change. looks good

**Comment by ScottSuarez** [comment]
superseded by https://github.com/GoogleCloudPlatform/magic-modules/pull/5117

**Pull request closed without merging**